Alignment/OfflineValidation/plugins/OverlapValidation.cc

```diff
@@ -0,0 +1,876 @@
+    else if ( (*iol).first->recHit()->geographicalId().subdetId()==1 ) layer_ =  layerFromId((*iol).first->recHit()->geographicalId().rawId(), tTopo)+20;
+    else if (  (*iol).first->recHit()->geographicalId().subdetId()==2 ) layer_ =  layerFromId((*iol).first->recHit()->geographicalId().rawId(), tTopo)+30; 
+    else layer_ = 99;
+    
+
+    if ( overlapIds_[0] ==  SiStripDetId((*iol).first->recHit()->geographicalId()).glued() )
+    edm::LogWarning("Overlaps") << "BAD GLUED: First Id = " << overlapIds_[0] << " has glued = " << SiStripDetId((*iol).first->recHit()->geographicalId()).glued() << "  and stereo = " << SiStripDetId((*iol).first->recHit()->geographicalId()).stereo() << endl;
+    if ( overlapIds_[1] ==  SiStripDetId((*iol).second->recHit()->geographicalId()).glued() )
+    edm::LogWarning("Overlaps") << "BAD GLUED: Second Id = " <<overlapIds_[1] << " has glued = " << SiStripDetId((*iol).second->recHit()->geographicalId()).glued() << "  and stereo = " << SiStripDetId((*iol).second->recHit()->geographicalId()).stereo() << endl;
+
+    const TransientTrackingRecHit::ConstRecHitPointer firstRecHit = (*iol).first->recHit();
+    const TransientTrackingRecHit::ConstRecHitPointer secondRecHit = (*iol).second->recHit();
+    hitPositions_[0] = firstRecHit->localPosition().x();
+    hitErrors_[0] = sqrt(firstRecHit->localPositionError().xx());
+    hitPositions_[1] = secondRecHit->localPosition().x();
+    hitErrors_[1] = sqrt(secondRecHit->localPositionError().xx());
+
+    hitPositionsY_[0] = firstRecHit->localPosition().y();
+    hitErrorsY_[0] = sqrt(firstRecHit->localPositionError().yy());
+    hitPositionsY_[1] = secondRecHit->localPosition().y();
+    hitErrorsY_[1] = sqrt(secondRecHit->localPositionError().yy());
+
+    //cout << "printing local X hit position and error for the overlap hits. Hit 1 = " << hitPositions_[0] << "+-" << hitErrors_[0] << "  and hit 2 is "  << hitPositions_[1] << "+-" << hitErrors_[1] << endl;
+
+    DetId id1 = (*iol).first->recHit()->geographicalId();
+    DetId id2 = (*iol).second->recHit()->geographicalId();
+    int layer1 = layerFromId(id1, tTopo);
+    int subDet1 = id1.subdetId();
+    int layer2 = layerFromId(id2, tTopo);
+    int subDet2 = id2.subdetId();
+    if (abs(hitPositions_[0])>5) edm::LogWarning("Overlaps") << "BAD: Bad hit position: Id = " << id1.rawId()  << " stereo = " << SiStripDetId(id1).stereo() << "  glued = " << SiStripDetId(id1).glued() << " from subdet = " << subDet1 << " and layer = " << layer1 << endl;
+    if (abs(hitPositions_[1])>5) edm::LogWarning("Overlaps") << "BAD: Bad hit position: Id = " << id2.rawId()  << " stereo = " << SiStripDetId(id2).stereo() << "  glued = " << SiStripDetId(id2).glued() << " from subdet = " << subDet2 << " and layer = " << layer2 << endl;
+    
+
+    // get track momentum
+    momentum_ = comb1.globalMomentum().mag();
+    
+    // get cluster size
+    if (subDet1>2) { //strip
```

Comment:
please use the 
`PixelSubdetector::PixelBarrel` and `PixelSubdetector::PixelEndcap` negation instead (or the OR of the 4 Strip subdetIds).

Reply:
Done